### PR TITLE
Fleshed out config file and black box

### PR DESCRIPTION
**Pull request by acpaquette**
This is going to need some form of testing at some point. I believe everything is point to the right places so a few runs with different parameters should help to flesh out any potential bugs.

### CODE REVIEW

**Comment by jlaura** [comment]
@all I am going to :+1: this, but would prefer that we have everyone take at least a quick scan.

File: bin/image_match.py [hunk 1]
```diff
@@ -12,40 +14,63 @@ def parse_arguments():
...
+        cg = CandidateGraph.from_filelist(find_in_dict(config_dict, 'inputfile_path') + args.input_file)
 
     # Apply SIFT to extract features
-    cg.extract_features(method='sift', extractor_parameters={'nfeatures': 1000})
+    cg.extract_features(method=find_in_dict(config_dict, 'extract_features')['method'],
+                        extractor_parameters=find_in_dict(config_dict, 'extractor_parameters'))
 
     # Match
-    cg.match_features()
+    cg.match_features(k=find_in_dict(config_dict, 'match_features')['k'])
```
Comment (Kelvinrr) [inline bin/image_match.py]: Is this good use of the `find_in_dict` function? Is this any different from `config_dict['match_features']['k']`

**Comment by kree** [comment]
I gave it a quick look, and all looks reasonable to me.

File: bin/image_match.py (hunk 1 above)
Comment (jlaura) [inline bin/image_match.py]: :+1: strange usage.

**Comment by acpaquette** [comment]
Don't pull this yet. For whatever reason no key points are being generated onto the images. I am looking into the details and trying to figure out why.

**Comment by jlaura** [comment]
@acpaquette What is the status on this?

**Comment by acpaquette** [comment]
@jlaura I haven't been able to work on it since we tried to update my system. I have uninstalled, cleaned, and reinstalled with no luck, I have also tried setting up a new environment which didn't work either. This is what the system is throwing.

(autocnet)bash-4.3$ python bin/image_match.py
Traceback (most recent call last):
  File "bin/image_match.py", line 10, in <module>
    from autocnet.graph.network import CandidateGraph
  File "/home/acpaquette/autocnet/autocnet/graph/network.py", line 11, in <module>
    from autocnet.fileio.io_gdal import GeoDataset
  File "/home/acpaquette/autocnet/autocnet/fileio/io_gdal.py", line 9, in <module>
    from osgeo import ogr
  File "/scratch/anaconda3/envs/autocnet/lib/python3.5/site-packages/osgeo/__init__.py", line 21, in <module>
    _gdal = swig_import_helper()
  File "/scratch/anaconda3/envs/autocnet/lib/python3.5/site-packages/osgeo/__init__.py", line 17, in swig_import_helper
    _mod = imp.load_module('_gdal', fp, pathname, description)
  File "/scratch/anaconda3/envs/autocnet/lib/python3.5/imp.py", line 242, in load_module
    return load_dynamic(name, filename, file)
  File "/scratch/anaconda3/envs/autocnet/lib/python3.5/imp.py", line 342, in load_dynamic
    return _load(spec)
ImportError: libpcre.so.3: cannot open shared object file: No such file or directory

**Comment by acpaquette** [comment]
This should be good to merge now!

**Comment by jlaura** [comment]
No need to squash - merged.  Please close the JIRA ticket.

**Pull request merged**